api/src/controllers/publications.js

```diff
@@ -168,27 +170,46 @@ async function getGoogleScholarPublications(req, res) {
   const resultLinks = await page.$$('.gsc_a_t a');
   let links = [];
 
-  for (let i = noOfDummyLinks; i < resultLinks.length; i++) {
-    links.push(await resultLinks[i].getAttribute('href'));
-  }
+  if (resultLinks.length === noOfDummyLinks) {
+    // no pubs found
+    endOfProfile = true;
+    await browser.close();
 
-  await browser.close();
+    response = {
+      retrieved: publications.length,
+      newPublications: [],
+      reachedEnd: endOfProfile,
+    };
+  } else {
+    for (let i = noOfDummyLinks; i < resultLinks.length; i++) {
+      links.push(await resultLinks[i].getAttribute('href'));
+    }
 
-  await Promise.all(links.map((x) => scrapeGoogleScholar(x))).then((pub) =>
-    publications.push(...pub)
-  );
+    await browser.close();
 
-  console.timeEnd('timeToScrape');
+    await Promise.all(links.map((x) => scrapeGoogleScholar(x))).then((pub) =>
+      publications.push(...pub)
+    );
 
-  const newPublications = await validateImportedPublications(
-    teamId,
-    publications
-  );
+    console.timeEnd('timeToScrape');
```

Comment:
Do we need to log timing info in prod? I'd suggest maybe enabling this only for dev/debug mode (e.g. through the use of environment variables, and if, for example: `process.env.DEBUG` evaluates to true, we can enable these log statements.

Reply:
After your comment, I'm thinking I'll probably get rid of them as a whole because playwright is looking pretty consistent and stable right now and (fingers crossed) it won't break anytime soon.

Reply:
SG